### PR TITLE
Refactor .NetCore API specific analyzers into a single analyzers package

### PR DESCRIPTION
**Pull request by mavasani**
Implements bullet (2) and (3) from https://github.com/dotnet/roslyn-analyzers/pull/1138#issue-220803935.

1. Now we have a single `Microsoft.NetCore.Analyzers` package for following analyzers:
   1. Microsoft.Composition.Analyzers
   2. System.Collection.Immutable.Analyzers
   3. System.Resources.Analyzers
   4. System.Runtime.Analyzers
   5. System.Runtime.InteropService.Analyzers
   6. System.Threading.Tasks.Analyzers
   7. System.Security.Cryptography.Hashing.Algorithms.Analyzers

2. `Desktop.Analyzers` has been renamed to `Microsoft.NetFramework.Analyzers`

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @jinujoseph @dotnet/analyzer-ioperation for review

File: src/Microsoft.CodeQuality.Analyzers/Core/Microsoft.CodeQuality.Analyzers.Core.csproj [hunk 1]
```diff
@@ -46,7 +46,7 @@
     <EmbeddedResource Update="Documentation\XmlDocumentationCommentsAnalyzersResources.resx">
       <Generator>ResXFileCodeGenerator</Generator>
       <LastGenOutput>XmlDocumentationCommentsAnalyzersResources.Designer.cs</LastGenOutput>
-      <CustomToolNamespace>Documentation\XmlDocumentationComments.Analyzers</CustomToolNamespace>
```
Comment (jinujoseph) [inline src/Microsoft.CodeQuality.Analyzers/Core/Microsoft.CodeQuality.Analyzers.Core.csproj]: intentional change ?

Comment (mavasani) [inline src/Microsoft.CodeQuality.Analyzers/Core/Microsoft.CodeQuality.Analyzers.Core.csproj]: Yes, the previous CustomToolNamespace was incorrect.

File: src/Microsoft.NetCore.Analyzers/Core/Microsoft.Composition.Analyzers/MicrosoftCompositionAnalyzersResources.Designer.cs [hunk 2]
```diff
@@ -40,7 +40,8 @@ internal class MicrosoftCompositionAnalyzersResources {
         internal static global::System.Resources.ResourceManager ResourceManager {
             get {
                 if (object.ReferenceEquals(resourceMan, null)) {
-                    global::System.Resources.ResourceManager temp = new global::System.Resources.ResourceManager("Microsoft.Composition.Analyzers.MicrosoftCompositionAnalyzersResources", typeof(MicrosoftCompositionAnalyzersResources).GetTypeInfo().Assembly);
+                    global::System.Resources.ResourceManager temp = new global::System.Resources.ResourceManager("Microsoft.NetCore.Analyzers.Microsoft.Composition.Analyzers.MicrosoftCompositionA" +
```
Comment (jinujoseph) [inline src/Microsoft.NetCore.Analyzers/Core/Microsoft.Composition.Analyzers/MicrosoftCompositionAnalyzersResources.Designer.cs]: Microsoft.NetCore.Analyzers.Microsoft.Composition.Analyzers.MicrosoftComposition... ?

File: src/Microsoft.CodeQuality.Analyzers/Core/Microsoft.CodeQuality.Analyzers.Core.csproj (hunk 1 above)
Comment (mavasani) [inline src/Microsoft.CodeQuality.Analyzers/Core/Microsoft.CodeQuality.Analyzers.Core.csproj]: Actually, I also need to regenerate the designer.cs file now that the custom tool namespace changed. I'll do it with a separate commit.

File: src/Microsoft.NetCore.Analyzers/Core/Microsoft.Composition.Analyzers/MicrosoftCompositionAnalyzersResources.Designer.cs (hunk 2 above)
Comment (mavasani) [inline src/Microsoft.NetCore.Analyzers/Core/Microsoft.Composition.Analyzers/MicrosoftCompositionAnalyzersResources.Designer.cs]: Yes, it seems to be auto-generated where it is appending the assembly name with custom tool namespace.

Comment (mavasani) [inline src/Microsoft.NetCore.Analyzers/Core/Microsoft.Composition.Analyzers/MicrosoftCompositionAnalyzersResources.Designer.cs]: Actually the custom tool is appending the directory name at the end of assembly name:
`<AssemblyName>\<DirectoryName>\<ResxName>`

File: src/Microsoft.CodeQuality.Analyzers/Core/Microsoft.CodeQuality.Analyzers.Core.csproj (hunk 1 above)
Comment (mavasani) [inline src/Microsoft.CodeQuality.Analyzers/Core/Microsoft.CodeQuality.Analyzers.Core.csproj]: I verified that regenerating the .designer.cs on my machine produces no diffs, so the custom tool was already stripping off characters before the last `\`

**Comment by mavasani** [comment]
Juju signed off on this offline

**Pull request merged**